xtas/tasks/es.py

```diff
@@ -100,32 +139,58 @@ def fetch_query_batch(idx, typ, query, field='body'):
     hits = scan(_es(), {'query': query}, index=idx, doc_type=typ)
     return [hit['_source'][field] for hit in hits if field in hit['_source']]
 
-CHECKED_MAPPINGS = set()
```

Comment:
Yes, the types remain while the cache gets flushed. I didn't write or profile this, but I imagine that doing a put_mapping for each document can get quite expensive.